src/main/connector/system/LinuxProcess/LinuxProcess.yaml

```diff
@@ -23,48 +23,61 @@ monitors:
       sources:
         process:
           type: commandLine
-          commandLine: ps -eo user,pid,pcpu,pmem,etimes,command
+          commandLine: /usr/bin/ps --no-headers -e -w -o pid,ppid,comm,user,pcpu,pmem,rss,vsz,thcount,etimes,command
           computes:
           - type: awk
             script: |
               {
-                user = $1
-                pid = $2
-                cpu = $3
-                mem = $4
-                time = $5
-                cpu_time = (cpu / 100) * time
-                # Concatenate the full command from $6 to $NF
+                pid = $1
+                ppid = $2
+                comm = $3
+                user = $4
+                cpu = $5
+                mem = $6
+                rss = $7 * 1024
+                vsz = $8 * 1024
+                thcount= $9
+                etimes = $10
+                cputimes = (cpu / 100) * etimes
```

Comment:
`pcpu` is not precise enough to calculate the CPU time for the process (the `etimes` value may be as large as 15,000,000 seconds for a process that's been running for a few months).

Also, `etimes` is a recent addition and is not available on older systems.

Let's use the `time` and `etime` then, but these are formatted as `[dd-]hh:mm:ss`, so it requires some additional processing.

The command is therefore:

```bash
/bin/ps --no-headers -e -w -o pid,ppid,comm,user,time,pmem,rss,vsz,thcount,etime,command
```

Then some AWK magic:

```awk
# [...]

cpuTime = $5
elapsedTime = $10

# [...] 

elementCount = split(cpuTime, cpuTimeElements, "[-:]");
if (elementCount == 3) {
  cpuSeconds = cpuTimeElements[1] * 3600 + cpuTimeElements[2] * 60 + cpuTimeElements[3];
} else if (elementCount == 4) {
  cpuSeconds = cpuTimeElements[1] * 3600 * 24 + cpuTimeElements[2] * 3600 + cpuTimeElements[3] * 60 + cpuTimeElements[4];
}
# Same to calculate elapsedSeconds
```

Reply:
In the OpenTelemetry documentation, they mention both metrics process.cpu.time and process.cpu.utilization.

So, should the command be /bin/ps --no-headers -e -w -o pid,ppid,comm,user,pcpu,time,pmem,rss,vsz,thcount,etime,command, or would you prefer to remove pcpu?